File: asa/education/report/student_report/student_report.py
# Copyright (c) 2023, Senituli Taumoepeau and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
from itertools import groupby
import json
import time
import math
import ast
import os.path
import sys
import datetime
import pandas as pd
import numpy as np
import functools

from openpyxl import load_workbook
from frappe import _, msgprint, utils
from datetime import datetime, timedelta
from frappe.utils import flt, getdate, datetime, comma_and
from collections import defaultdict
from werkzeug.wrappers import Response
import frappe


def execute(filters=None):
	if not filters: filters = {}
	columns, data = [], []

	termSQL = filters.get("academic_term")
	yearSQL = filters.get("academic_year")
	studentSQL = filters.get("student")
	studentsql = "tabAR.student = '{student}'".format(student=filters.student) if filters.student else "1 = 1"

	attendanceSQL = frappe.db.sql("""SELECT student, student_name, date, academic_term, status 
					FROM `tabStudent Attendance`
					WHERE student = %s
					AND YEAR(date) = %s
					AND academic_term = %s""", (studentSQL, yearSQL, termSQL), as_dict = 1)
	

	dataframe = pd.DataFrame.from_records(attendanceSQL)


	return columns, data

File: asa/education/report/student_report/student_report.js
// Copyright (c) 2023, Senituli Taumoepeau and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Student Report"] = {
	"filters": [
		{
			"fieldname":"academic_term",
			"label":__("Academic Term"),
			"fieldtype":"Link",
			"options": "Academic Term",
			"width": "90px",
			"reqd": 1
		},,
		{
			"fieldname":"academic_year",
			"label":__("Academic Year"),
			"fieldtype":"Link",
			"options": "Academic Year",
			"width": "90px",
			"reqd": 1
		},
		{
			"fieldname": "student",
			"label": __("Student ID"),
			"fieldtype": "Link",
			"options": "Student",
			"reqd": 1,
			on_change: () => {
				var student = frappe.query_report.get_filter_value('student');			
				var academic_term = frappe.query_report.get_filter_value('academic_term');
				var academic_year = frappe.query_report.get_filter_value('academic_year');

				if (student) {
					frappe.db.get_value('Student', student, ["title"], function(value) {
					frappe.query_report.set_filter_value('title', value["title"].toUpperCase());
					});
				}
			}
		},
		{
			"fieldname": "title",
			"label": __("Student Name"),
			"fieldtype": "Data",
			"hidden": 1
		},

	]
};
